src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseTransferService.java

```diff
@@ -0,0 +1,141 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ecm.common.client.CcdClient;
+import uk.gov.hmcts.ecm.common.exceptions.CaseCreationException;
+import uk.gov.hmcts.ecm.common.model.ccd.CaseData;
+import uk.gov.hmcts.ecm.common.model.ccd.CaseDetails;
+import uk.gov.hmcts.ecm.common.model.ccd.SubmitEvent;
+import uk.gov.hmcts.ecm.common.model.ccd.items.BFActionTypeItem;
+import uk.gov.hmcts.ecm.common.model.ccd.items.DateListedTypeItem;
+import uk.gov.hmcts.ecm.common.model.ccd.items.HearingTypeItem;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.*;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service("caseTransferService")
+public class CaseTransferService {
+
+    private final PersistentQHelperService persistentQHelperService;
+    private final CcdClient ccdClient;
+    private static final String MESSAGE = "Failed to retrieve the case for case id : ";
+
+    @Value("${ccd_gateway_base_url}")
+    private String ccdGatewayBaseUrl;
+
+    private CaseData getOriginalCase(CaseDetails caseDetails, String userToken) {
+        try {
+            CaseData caseData = caseDetails.getCaseData();
+            if (caseData.getCounterClaim() != null && !caseData.getCounterClaim().trim().isEmpty()) {
```

Comment:
The google guava library is include in the project. It has some handy utils for this sort of problem.
See `Strings.isNullOrEmpty()`

Reply:
done